api/api.js

```diff
@@ -0,0 +1,147 @@
+require('dotenv').config();
+/**
+ * third party libraries
+ */
+const cors = require('cors');
+const http = require('http');
+const helmet = require('helmet');
+const express = require('express');
+const bodyParser = require('body-parser');
+const morgan = require('morgan')('combined');
+const { execute, subscribe } = require('graphql');
+const { pubsub } = require('./subscriptions');
+const mapRoutes = require('express-routes-mapper');
+const authUserService = require('./services/auth.user.service');
+const { graphqlExpress } = require('apollo-server-express');
+const { SubscriptionServer } = require('subscriptions-transport-ws');
+const expressPlayground = require('graphql-playground-middleware-express')
+  .default;
+
+/**
+ * server configuration
+ */
+const config = require('../config/');
+const schema = require('./controllers/');
+const auth = require('./policies/auth.policy');
+const dbService = require('./services/db.service');
+const authService = require('./services/auth.service');
+const environment = process.env.NODE_ENV;
+
+/**
+ * Custome middleware
+ */
+const authorization = require('./middlewares/authorization');
+
+/**
+ * express application
+ */
+const api = express();
+const server = http.Server(api);
+const mappedRoutes = mapRoutes(config.publicRoutes, 'api/controllers/Auth/');
+const DB = dbService(environment, config.migrate).start();
+
+// allow cross origin requests
+// configure to allow only requests from certain origins
+api.use(cors());
+api.use(morgan);
+
+// secure express app
+api.use(
+  helmet({
+    dnsPrefetchControl: false,
+    frameguard: false,
+    ieNoOpen: false,
+  })
+);
+
+//Get the user from the JWT if there is one
+api.use(authorization);
+
+// parsing the request bodys
+api.use(bodyParser.urlencoded({ extended: false }));
+api.use(bodyParser.json());
+
+// public REST API
+api.use('/', mappedRoutes);
+
+// private GraphQL API
+api.all('/graphql');
+api.use(
+  '/graphql',
+  bodyParser.json(),
+  graphqlExpress(req => ({
+    schema,
+    context: { auth: req.auth },
+    cacheControl: true,
+  }))
+);
+
+//Playground tool for GraphQL
+api.get(
+  '/explore',
+  expressPlayground({
+    endpoint: '/graphql',
+    subscriptionsEndpoint: 'ws://localhost:2017/subscriptions',
+  })
+);
+
+server.listen(config.port, () => {
+  new SubscriptionServer(
```

Comment:
The content of this callback is too coupled to graphql behavior, rather than general api. I mean I'll put it in a graphql folder or something like that.